Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp

```diff
@@ -135,31 +139,101 @@ class AcceleratedBufferPool {
     RunLoop::Timer m_releaseUnusedBuffersTimer;
 };
 
+class UnacceleratedThreadedRenderingPool {
+    WTF_MAKE_FAST_ALLOCATED();
+public:
+    static UnacceleratedThreadedRenderingPool& singleton()
```

Comment:
I'm not sure this singleton is a good idea (and the same applies to the accelerated buffer poll). What happens if the web page is closed and there are still ongoing works? The WorkerPool automatically finishes the threads in the destructor, but it's never called in this case. In nicosia implementation, the painting engines is owned by the compositing coordinator. I'm not sure that's the right place either, but for now we could probably do the same.

Reply:
For now, CompositingCoordinator seems to best place to manage the lifetime of the AcceleratedBufferPool & co - I will rename them to SkiaAcceleratedBufferPool/SkiaUnacceleratedThreadedRenderingPool, move them to their own cpp/h files and remove the singleton. They can be simply instances held by CompositingCoordinator.